front/lib/api/assistant/generation.ts

```diff
@@ -184,6 +185,42 @@ export async function renderConversationForModel({
     const c = r.value;
     if (tokensUsed + c <= allowedTokenCount) {
       tokensUsed += c;
+      selected.unshift(messages[i]);
+    } else if (messages[i].role === "content_fragment") {
+      const c = allowedTokenCount - tokensUsed;
+      tokensUsed += c;
+      const contentFragmentMessage =
```

Comment:
I would rather copy that information back into ModelMessageType to avoid going back to that array in this logic?

Reply:
ok but we need to strip it out after because otherwise it's sent as an input to the Dust App.